Tasks/AzureResourceManagerTemplateDeploymentV3/operations/DeploymentScopeBase.ts

```diff
@@ -0,0 +1,92 @@
+import tl = require("azure-pipelines-task-lib/task");
+
+import deployAzureRG = require("../models/DeployAzureRG");
+import armResource = require("azure-arm-rest-v2/AzureServiceClientBase");
+import utils = require("./Utils");
+import { sleepFor } from 'azure-arm-rest-v2/webClient';
+import { DeploymentParameters } from "./DeploymentParameters";
+
+export class DeploymentScopeBase {
+    protected deploymentParameters: DeploymentParameters;
+    protected taskParameters: deployAzureRG.AzureRGTaskParameters;
+    protected armClient: armResource.AzureServiceClientBase;
+
+    constructor(armClient: armResource.AzureServiceClientBase, taskParameters: deployAzureRG.AzureRGTaskParameters, deploymentParameters?: DeploymentParameters) {
+        this.taskParameters = taskParameters;
+        this.armClient = armClient;
+        this.deploymentParameters = deploymentParameters;
+    }
+
+    protected async createTemplateDeployment() {
+        console.log(tl.loc("CreatingTemplateDeployment"));
+        var params: DeploymentParameters;
+        if (this.taskParameters.templateLocation === "Linked artifact") {
+            params = utils.getDeploymentDataForLinkedArtifact(this.taskParameters);
+        } else if (this.taskParameters.templateLocation === "URL of the file") {
+            params = await utils.getDeploymentObjectForPublicURL(this.taskParameters);
+        } else {
+            throw new Error(tl.loc("InvalidTemplateLocation"));
+        }
+
+        if(!!this.deploymentParameters){
+            params.location = this.deploymentParameters.location;
+        }
+
+        this.deploymentParameters = params;
+        await this.performAzureDeployment(3);
+    }
+
+    protected async performAzureDeployment(retryCount = 0): Promise<void> {
+        if (this.deploymentParameters.properties["mode"] === "Validation") {
+            return this.validateDeployment();
+        } else {
+            console.log(tl.loc("StartingDeployment"));
+            return new Promise<void>((resolve, reject) => {
+                this.taskParameters.deploymentName = this.taskParameters.deploymentName || utils.createDeploymentName(this.taskParameters);
+                console.log(tl.loc("LogDeploymentName", this.taskParameters.deploymentName));
+                this.armClient.deployments.createOrUpdate(this.taskParameters.deploymentName, this.deploymentParameters, (error, result, request, response) => {
```

Comment:
should we print the intermittent status while deployment is going on. we can print a list of resources which have been created till now

Reply:
Is this a feedback or any user's request ?